src/components/ConditionsLayer/Flow/Flow.tsx

```diff
@@ -0,0 +1,235 @@
+import { MouseEvent, useCallback, useEffect, useState } from "react";
+import ReactFlow, {
+  Background,
+  Controls,
+  Edge,
+  Node,
+} from "react-flow-renderer";
+import ConditionNode, {
+  ConditionNodeTypes,
+} from "components/ConditionsLayer/Nodes/ConditionNode";
+import DashedEdge from "components/Edges/DashedEdge";
+import {
+  EdgeTypes,
+  initialEdges,
+  initialNodes,
+  NodeTypes,
+} from "./initialElements";
+import StartEndNode from "../Nodes/StartEndNode";
+import { StartEndNodeTypes } from "../Nodes/StartEndNode/StartEndNode";
+
+const nodeTypes = { condition: ConditionNode, startEnd: StartEndNode };
+const edgeTypes = {
+  dashed: DashedEdge,
+};
+
+interface FlowProps {
+  arePlaceholdersVisible: boolean;
+}
+
+const findLastColumn = (nodes: Node[]) => {
+  let lastCol = 0;
+  nodes.map(({ data }) => (data.col > lastCol ? (lastCol = data.col) : null));
+  return lastCol;
+};
+
+interface ColumnData {
+  col: number;
+  row: number;
+  id: string | null;
+}
+
+const findLastRow = (nodes: Node[]) => {
+  let maxRow = 0;
+  nodes.map(({ data }) => (data.row > maxRow ? (maxRow = data.row) : null));
+  return maxRow;
+};
+
+const lastElementInEachRow = (nodes: Node[]) => {
+  let colArr: ColumnData[] = [];
+  const preparedNodes = nodes.filter(
+    (item) => item.data.type !== ConditionNodeTypes.PLACEHOLDER
+  );
+
+  preparedNodes.map(({ data }) => {
+    if (!colArr.find((item) => item.row === data.row))
+      colArr.push({ col: 0, row: data.row, id: null });
+  });
+
+  colArr = colArr.map(({ row }) => {
+    let colMax = 0;
+    let tempId = "";
+    for (const { id, data } of preparedNodes) {
+      if (row === data.row && data.col >= colMax) {
+        colMax = data.col;
+        tempId = id;
+      }
+    }
+    return {
+      row,
+      col: colMax,
+      id: tempId,
+    };
+  });
+
+  return colArr;
+};
+
+const Flow = ({ arePlaceholdersVisible }: FlowProps) => {
+  const [nodes, setNodes] = useState<Node[]>(initialNodes);
+  const [edges, setEdges] = useState<Edge[]>(initialEdges);
+  const [filtratedNodes, setFiltretedNodes] = useState<Node[]>(initialNodes);
```

Comment:
nitpick: filtrated and filtreted looks like a typo. Not sure if it's a good practice, but I usually see the getter and setter having the same name. `filtered/setFiltered` is also a good option here :)